### ISSUE TITLE
Warning from Visual Studio 19 regarding unsafe use of bool

### ISSUE THREAD

**Issue by bjornblissing** [google/draco#814]
The following section spawns a C4804 warning when the template type `T` is `bool`:
```
geometry_attribute.h(290,13): warning C4804: '<': unsafe use of type 'bool' in operation
geometry_attribute.h(290,35): warning C4804: '>': unsafe use of type 'bool' in operation
```

https://github.com/google/draco/blob/9522c22a74c6734cdeb29ca1b6c55a1047ef0bae/src/draco/attributes/geometry_attribute.h#L287-L293

This warning can be avoided by skipping the check when type `T` is `bool`. This check can be skipped safely since we are guaranteed that `bool` will fit in the output type `OutT`, given that `OutT` is an integral type.

Suggested change on Line 287:
```cpp
if (!std::is_same<T, bool>::value && std::is_integral<T>::value && std::is_integral<OutT>::value) {
```

**Comment by tomfinegan** [comment]
Please send a PR if you'd like to see this change merged into the repo.

**Comment by bjornblissing** [comment]
I have submitted a PR #817

**Comment by tomfinegan** [comment]
Fixed by #817. 

@bjornblissing  Thanks for the PR!
